distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering/Natural_Order.enso

```diff
@@ -17,62 +20,61 @@ import Standard.Base.Data.Ordering.Vector_Lexicographic_Order
          ["a2", "a1", "a100", "a001", "a0001"].sort by=Natural_Order.compare . should_equal ["a0001", "a001", "a1", "a2", "a100"]
 compare : Text -> Text -> Ordering
 compare text1 text2 =
-    ## TODO [RW] A more efficient algorithm which works better for longer texts
-       should be implemented, as described in
-       https://www.pivotaltracker.com/story/show/181176589
-    nat1 = here.to_natural_key text1
-    nat2 = here.to_natural_key text2
-    ## TODO [RW] The additional assignment to a temporary variable is a
-       workaround for the following bug:
-       https://www.pivotaltracker.com/story/show/181162108
-    res = Vector_Lexicographic_Order.compare nat1 nat2
-    res
-
-## PRIVATE
-to_natural_key : Text -> Vector
-to_natural_key text = if text.is_empty then [] else
-    splitter_regex = Regex.compile "([^0-9]+|[0-9]+)"
-    parts = splitter_regex.find text mode=Mode.All
-
-    ## TODO [RW] Currently there is no `is_digit` method. Once
-       https://www.pivotaltracker.com/story/show/181176532 is implemented, this
-       regex can be replaced with a simpler and faster `is_digit` check.
-    is_integer_regex = Regex.compile "[0-9]+"
-    parts.map part-> case is_integer_regex.matches part of
-        True ->
-            ## TODO [RW] Currently there is no `Integer.parse` method, so we
+    iter1 = BreakIterator.getCharacterInstance
+    iter1.setText text1
+
+    iter2 = BreakIterator.getCharacterInstance
+    iter2.setText text2
+
+    is_digit=c -> c>=48 && c<=57
+
+    get_number text prev next iter =
+        find_number text next iter =
+            new_next = iter.next
+            if (new_next == -1) then -next else
+                substring = Text_Utils.substring text next new_next
+                c = Text_Utils.get_chars substring . at 0
+                if (is_digit c).not then next else
+                    @Tail_Call find_number text new_next iter
+
+        n = find_number text next iter
+        s = Text_Utils.substring text prev n.abs
```

Comment:
IMO we should not use 1-letter var names. They are hard to understand and make code hard to maintain. Please, find always a meaningful, semantic names for variables – they are far more superior than anything else for the code quality and readability :)